core/include/sched.h

```diff
@@ -188,6 +188,25 @@ void sched_register_cb(void (*callback)(uint32_t, uint32_t));
 
 #endif
 
+#ifdef BOARD_NATIVE
```

Comment:
I think the problem is even more complicated, since the `pthread.h` from the host system requires the defines/types in a provided `sched.h`. So they must be available in the header, directly like currently or indirectly by include a further header.